annotations/transport/annotations_router.go

```diff
@@ -234,3 +241,42 @@ func decodeUpdateAnnotationRequest(r *http.Request) (*influxdb.AnnotationCreate,
 
 	return u, nil
 }
+
+func storedAnnotationToEvent(s *influxdb.StoredAnnotation) (*influxdb.AnnotationEvent, error) {
+	st, err := tStringToPointer(s.Lower)
+	if err != nil {
+		return nil, err
+	}
+
+	et, err := tStringToPointer(s.Upper)
+	if err != nil {
+		return nil, err
+	}
+
+	return &influxdb.AnnotationEvent{
+		ID: s.ID,
+		AnnotationCreate: influxdb.AnnotationCreate{
+			StreamTag: s.StreamTag,
+			Summary:   s.Summary,
+			Message:   s.Message,
+			Stickers:  stickerSliceToMap(s.Stickers),
+			EndTime:   et,
+			StartTime: st,
+		},
+	}, nil
+
+}
+
+func stickerSliceToMap(stickers []string) map[string]string {
+	stickerMap := map[string]string{}
+
+	for i := range stickers {
+		sticks := strings.SplitN(stickers[i], "=", 2)
+		if len(sticks) < 2 {
+			continue
```

Comment:
Should we make this an error case, or log it? I worry about silently ignoring entries if they're malformatted will cause headaches later.

Reply:
I basically just lifted this code from IDPE, but I do agree about silently ignoring errors being bad so I went ahead and made it return and error and added some more tests cases for it.